ui/app/src/modules/Content/Dependencies/DependenciesDialogUI.tsx

```diff
@@ -0,0 +1,473 @@
+                <Select
+                  value={state.selectedOption}
+                  onChange={(event: React.ChangeEvent<{ value: unknown }>) => {
+                    setState({ selectedOption: event.target.value as string });
+                  }}
+                  inputProps={{
+                    className: classes.select
+                  }}
+                >
+                  <MenuItem value='depends-on'>
+                    <FormattedMessage
+                      id="dependenciesDialog.dependsOn"
+                      defaultMessage="Items that depend on selected item"
+                    />
+                  </MenuItem>
+                  <MenuItem value='depends-on-me'>
+                    <FormattedMessage
+                      id="dependenciesDialog.dependsOnMe"
+                      defaultMessage="Dependencies of selected item"
+                    />
+                  </MenuItem>
+                </Select>
+              </FormControl>
+            </div>
+            <SuspenseWithEmptyState
+              resource={resource}
+              withEmptyStateProps={{
+                emptyStateProps: {
+                  title: (
+                    <FormattedMessage
+                      id="dependenciesDialog.emptyStateMessage"
+                      defaultMessage="No Dependencies found"
+                    />
+                  ),
+                  classes: {
+                    root: classes.suspense
+                  }
+                }
+              }}
+              loadingStateProps={{
+                classes: {
+                  root: classes.suspense
+                }
+              }}
+            >
+              <DependenciesList
+                resource={resource}
+                state={state}
+                setState={setState}
+                assetsTypes={assetsTypes}
+                isEditableItem={isEditableItem}
+                handleEditorDisplay={handleEditorDisplay}
+              />
+            </SuspenseWithEmptyState>
+          </DialogBody>
+          <DialogFooter
+            classes={{
+              root: classes.dialogFooter
+            }}
+          >
+            <FormControlLabel
+              className={classes.compactViewAction}
+              control={
+                <Checkbox
+                  checked={state.compactView}
+                  onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                    setState({ compactView: event.target.checked });
+                  }}
+                  color="primary"
+                />
+              }
+              label="Compact"
+            />
+            <FormControl className={classes.formControl}>
+              <Select
+                value={state.showTypes}
+                onChange={(event: React.ChangeEvent<{ value: unknown }>) => {
+                  setState({ showTypes: event.target.value as string });
+                }}
```

Comment:
if you set value to "string" instead of unknown, doesn't that solve the need to cast?

Reply:
If I set the value to string then I get this error:
`Types of property 'value' are incompatible.               Type 'unknown' is not assignable to type 'string'`